lib/rules/no-extra-parens.js

```diff
@@ -326,7 +326,7 @@ module.exports = function(context) {
 
         "ArrowFunctionExpression": function(node) {
             if (node.body.type !== "BlockStatement") {
-                if (node.body.type !== "ObjectExpression" && hasExcessParens(node.body) && precedence(node.body) >= precedence({type: "AssignmentExpression"})) {
+                if (sourceCode.getFirstToken(node.body).value !== "{" && hasExcessParens(node.body) && precedence(node.body) >= precedence({type: "AssignmentExpression"})) {
```

Comment:
I'm not sure what the precedence part is supposed to warn on, but it's not covered by any test (I know because I was also implementing this).